test/over_react/component_declaration/redux_component_test.dart

```diff
@@ -0,0 +1,124 @@
+    );
+    baseState = new BaseState();
+    baseActions = new BaseActions();
+  });
+
+  tearDown(() {
+    baseReducerBuilder = null;
+    baseState = null;
+    baseActions = null;
+  });
+
+  group('ReduxUiProps', () {
+    test('exposes a store getter', () {
+      var props = TestDefault();
+      var baseStore = new Store<BaseState, BaseStateBuilder, BaseActions>(
+        baseReducerBuilder.build(),
+        baseState,
+        baseActions,
+      );
+
+      props.store = baseStore;
+
+      expect(props.store, equals(baseStore));
+      expect(props, containsPair('TestDefaultProps.store', baseStore),
+          reason: 'should have a `store` getter with the a properly-namespaced prop key');
+    });
+  });
+
+  group('ReduxUiComponent', () {
+    test('subscribes to a single store by default', () async {
+      var store = new Store<BaseState, BaseStateBuilder, BaseActions>(
+        baseReducerBuilder.build(),
+        baseState,
+        baseActions,
+      );
+      var renderedInstance = render(TestDefault()..store = store);
+      TestDefaultComponent component = getDartComponent(renderedInstance);
+
+      store.actions.trigger1();
+      await new Future.delayed(Duration.ZERO);
+
+      expect(component.numberOfRedraws, 1);
+
+      unmount(renderedInstance);
+
+      store.actions.trigger1();
+      await new Future.delayed(Duration.ZERO);
+
+      expect(component.numberOfRedraws, 1,
+          reason: 'component should no longer be listening after unmount');
+    });
+
+    test('subscribes to any state changes subscribed to in connect', () async {
+      var stores = new Store<BaseState, BaseStateBuilder, BaseActions>(
+        baseReducerBuilder.build(),
+        baseState,
+        baseActions,
+      );
+      var renderedInstance = render(TestConnect()..store = stores);
+      TestConnectComponent component = getDartComponent(renderedInstance);
+
+      stores.actions.trigger1();
+      await new Future.delayed(Duration.ZERO);
+      expect(component.numberOfRedraws, 1);
+
+      stores.actions.trigger2();
+      await new Future.delayed(Duration.ZERO);
+      expect(component.numberOfRedraws, 1);
+    });
+
+    test('updates subscriptions when new props are passed', () async {
+      var store = new Store<BaseState, BaseStateBuilder, BaseActions>(
+        baseReducerBuilder.build(),
+        baseState,
+        baseActions,
+      );
+      var updatedStore = new Store<BaseState, BaseStateBuilder, BaseActions>(
+        baseReducerBuilder.build(),
+        new BaseState(),
+        new BaseActions(),
+      );
+      var renderedInstance = render((TestDefault()..store = store)());
+      TestDefaultComponent component = getDartComponent(renderedInstance);
+
+      store.actions.trigger1();
+      await new Future.delayed(Duration.ZERO);
+      expect(component.numberOfRedraws, 1);
+
+      component.props = TestDefault()..store = updatedStore;
+      component.redraw();
```

Comment:
This should be actually rerendering the component and not just simulating a props update.